bioio_ome_zarr/writers/ome_zarr_writer.py

```diff
@@ -177,8 +194,17 @@ def __init__(
 
         # 5) formatting and compression
         self.zarr_format = zarr_format
-        self.shard_factor = shard_factor
         self.compressor = compressor
+        self.shards_per_level: Optional[List[Tuple[int, ...]]] = None
+        if shard_shape is not None:
+            self.shards_per_level = _normalize_levelwise(
+                shard_shape,
+                num_levels=self.num_levels,
+                ndim=self.ndim,
+                label="shard_shape",
+            )
```

Comment:
there should also be an error check that shard shape values are exact multiples of chunk shape values.  This is in the zarr docs somewhere.